synapse/datamodel.py

```diff
@@ -127,6 +127,10 @@ def __init__(self, modl, form, name, typedef, info):
         self.deprecated = self.info.get('deprecated', False)
 
         self.type = self.modl.getTypeClone(typedef)
+        self.typehash = self.type.typehash
+
+        if self.type.isarray:
+            self.arraytypehash = self.type.arraytype.typehash
```

Comment:
should `arraytypehash` default to `None` on these types?

Reply:
We don't default `arraytype` to None on non-array types so I was going to leave it off, but I don't have a strong opinion on it either way.

Reply:
lets leave it as is for now. it should be as safe as the existing code :)